tests/models/beit/test_modeling_tf_beit.py

```diff
@@ -0,0 +1,638 @@
+                        list_input[index] = prepared_for_class[value]
+
+                    tuple_input = tuple(list_input)
+
+                    # Send to model
+                    loss = model(tuple_input[:-1])[0]
+
+                    self.assertEqual(loss.shape, [loss_size])
+
+    @slow
+    def test_model_from_pretrained(self):
+        for model_name in TF_BEIT_PRETRAINED_MODEL_ARCHIVE_LIST[:1]:
+            model = TFBeitModel.from_pretrained(model_name, from_pt=True)
+            self.assertIsNotNone(model)
+
+
+# We will verify our results on an image of cute cats
+def prepare_img():
+    image = Image.open("./tests/fixtures/tests_samples/COCO/000000039769.png")
+    return image
+
+
+@require_tf
+@require_vision
+class TFBeitModelIntegrationTest(unittest.TestCase):
+    @cached_property
+    def default_image_processor(self):
+        return (
+            BeitImageProcessor.from_pretrained("microsoft/beit-base-patch16-224-pt22k", from_pt=True)
+            if is_vision_available()
+            else None
+        )
+
+    @slow
+    def test_inference_masked_image_modeling_head(self):
+        model = TFBeitForMaskedImageModeling.from_pretrained("microsoft/beit-base-patch16-224-pt22k", from_pt=True)
+
+        image_processor = self.default_image_processor
+        image = prepare_img()
+        pixel_values = image_processor(images=image, return_tensors="tf")
+
+        # prepare bool_masked_pos
+        bool_masked_pos = tf.ones((1, 196), dtype=tf.dtypes.bool)
+
+        # forward pass
+        outputs = model(pixel_values=pixel_values, bool_masked_pos=bool_masked_pos)
+        logits = outputs.logits
+
+        # verify the logits
+        expected_shape = tf.convert_to_tensor([1, 196, 8192])
+        self.assertEqual(logits.shape, expected_shape)
+
+        expected_slice = tf.convert_to_tensor(
+            [[-3.2437, 0.5072, -13.9174], [-3.2456, 0.4948, -13.9401], [-3.2033, 0.5121, -13.8550]]
+        )
+
+        tf.debugging.assert_near(logits[:3, :3], expected_slice, atol=1e-4)
+
+    @slow
+    def test_inference_image_classification_head_imagenet_1k(self):
+        model = TFBeitForImageClassification.from_pretrained("microsoft/beit-base-patch16-224-pt22k", from_pt=True)
+
+        image_processor = self.default_image_processor
+        image = prepare_img()
+        inputs = image_processor(images=image, return_tensors="tf")
+
+        # forward pass
+        outputs = model(**inputs)
+        logits = outputs.logits
+
+        # verify the logits
+        expected_shape = tf.convert_to_tensor([1, 1000])
+        self.assertEqual(logits.shape, expected_shape)
+
+        expected_slice = tf.convert_to_tensor([0.3277, -0.1395, 0.0911])
+
+        tf.debugging.assert_near(logits[0, :3], expected_slice, atol=1e-4)
+
+        expected_top2 = [model.config.label2id[i] for i in ["remote control, remote", "tabby, tabby cat"]]
+        self.assertEqual(tf.nn.top_k(outputs.logits[0], 2).indices.numpy().tolist(), expected_top2)
```

Comment:
This should be updated to match the equivalent test in `test_modeling_beit.py`. Specifically: the checkpoint, expected values and the final assert